src/actions/sources/select.js

```diff
@@ -162,21 +164,22 @@ export function selectSpecificLocation(location: Location) {
       dispatch(closeActiveSearch());
     }
 
-    dispatch(addTab(source.toJS(), 0));
+    const source = sourceRecord.toJS();
 
+    dispatch(addTab(source, 0));
     dispatch({
       type: "SELECT_SOURCE",
-      source: source.toJS(),
+      source,
```

Comment:
I think I'd prefer `source: sourceRecord.toJS()` so it is clear why we're doing the serialization